pandas/core/indexes/period.py

```diff
@@ -508,7 +508,11 @@ def get_loc(self, key, method=None, tolerance=None):
             # _get_string_slice will handle cases where grp < freqn
             assert grp >= freqn
 
-            if grp == freqn:
+            # BusinessDay is a bit strange. It has a *lower* code, but we never parse
```

Comment:
any way to make this more generic here cc @jbrockmendel

Reply:
Yah its a bit wonky.  I think it might be nicer if done in terms of Resolution, but we'll need to implementing a mapping between FreqGroup and Resolution for that